packages/abi-coder/src/function-fragment.ts

```diff
@@ -0,0 +1,224 @@
+import type { BytesLike } from '@ethersproject/bytes';
+import { arrayify, concat } from '@ethersproject/bytes';
+import { Logger } from '@ethersproject/logger';
+import { sha256 } from '@ethersproject/sha2';
+import { bufferFromString } from '@fuel-ts/keystore';
+import { bn } from '@fuel-ts/math';
+import { versions } from '@fuel-ts/versions';
+
+import { AbiCoder } from './abi-coder';
+import type { DecodedValue, InputValue } from './coders/abstract-coder';
+import type { ArrayCoder } from './coders/array';
+import { TupleCoder } from './coders/tuple';
+import type { U64Coder } from './coders/u64';
+import { arrayRegEx, enumRegEx, OPTION_CODER_TYPE, stringRegEx, structRegEx } from './constants';
+import type {
+  JsonAbi,
+  JsonAbiArgument,
+  JsonAbiFunction,
+  JsonAbiFunctionAttribute,
+} from './json-abi';
+import { isPointerType, getVectorAdjustments } from './utilities';
+
+const logger = new Logger(versions.FUELS);
+
+export class FunctionFragment<
+  TAbi extends JsonAbi = JsonAbi,
+  FnName extends TAbi['functions'][number]['name'] = string
+> {
+  readonly signature: string;
+  readonly selector: string;
+  readonly name: string;
+  readonly jsonFn: JsonAbiFunction;
+  readonly attributes: readonly JsonAbiFunctionAttribute[];
+
+  private readonly jsonAbi: JsonAbi;
+  constructor(abi: JsonAbi, name: FnName) {
+    this.jsonAbi = abi;
+    this.jsonFn = abi.functions.find((f) => f.name === name)!;
```

Comment:
The linter has left these `Forbidden non-null assertion` warnings in several places on this PR. Can you have a look at them?

Reply:
This is due to the nature of `.find()` which returns `T | undefined`. I just realized that I can do an `as T` cast, so I'll do that.